### ISSUE TITLE
Comments blocks backports and fixes: Tracking issue

### ISSUE THREAD

**Issue by SantosGuillamot** [WordPress/gutenberg#41451]
Supersedes #34994 

The goal of this issue is to list and keep track of some issues related to the Comments blocks that are planned to be included in 6.0.1. For full context about the Comments blocks, please take a look at the original [tracking issue](https://github.com/WordPress/gutenberg/issues/34994).

There is one PR that has already been merged but wasn't included in 6.0 so it needs a backport:

- [x]  Backport [Refactor: Reuse code of the editor placeholder across Post Comments and Post Comments Form](https://href.li/?https://github.com/WordPress/gutenberg/pull/40560).

And some other fixes that makes sense to work on:

- [x] [Comments translation issue](https://github.com/WordPress/gutenberg/issues/41292).
- [x] [Warning i18n issue](https://github.com/WordPress/gutenberg/pull/42131).
- [x] [Post Comments Form: Add button that enables commenting to warning](https://github.com/WordPress/gutenberg/pull/41603)
- [x] [Comment Author and Date blocks: aligned editor markup with the frontend](https://github.com/WordPress/gutenberg/pull/41631)
- [x] [Comments Block: fixed issue with custom font sizes and links color](https://github.com/WordPress/gutenberg/pull/41627)

_**Update**: The following issues have been postponed to WP 6.1 (see https://github.com/WordPress/gutenberg/issues/41451#issuecomment-1165821623):_

- [x] [Merge Comments Query Loop block into Post Comments block](https://github.com/WordPress/gutenberg/issues/40521) (#41807).
- [x] Change the name and ID to Comments and `core/comments`.
  - [x] Name (https://github.com/WordPress/gutenberg/pull/40763)
  - [x] ID (https://github.com/WordPress/gutenberg/pull/40506)

**Comment by ockham** [comment]
We might wanna add [this issue](https://github.com/WordPress/gutenberg/issues/41308) which already has an in-progress [PR](https://github.com/WordPress/gutenberg/pull/41354).

**Comment by SantosGuillamot** [comment]
Thanks for sharing 🙂 Although it is related to the comments, the Avatar is a global block. Should we include it here anyway or it is better to handle it independently? Any thoughts? I don't have a strong opinion here.

**Comment by ockham** [comment]
> Thanks for sharing 🙂 Although it is related to the comments, the Avatar is a global block. Should we include it here anyway or it is better to handle it independently? Any thoughts? I don't have a strong opinion here.

Since it's kind of prominently used in the Comments blocks (and impacted by that bug), I think it makes sense if we make sure a fix finds its way into 6.0.1 🙂

**Comment by ockham** [comment]
Here's another PR we should include: https://github.com/WordPress/gutenberg/pull/41631

It's been added to the [6.0.1 project board](https://github.com/orgs/WordPress/projects/31), and I've added the "Backport to WP Minor Release" label. We might not need to edit our list every time to include issues/PRs like that (since they're covered by that project board and label), and instead use the list as our team's TODO for 6.0.1 🙂

**Comment by ockham** [comment]
As mentioned at https://github.com/WordPress/gutenberg/pull/41807#issuecomment-1160449503, it's probably a good idea to tackle the ID change (`core/comments-query-loop`) separately from the Comments/Post Comments block merge.

I've thus added https://github.com/WordPress/gutenberg/pull/40506 to the WP 6.0.1 board.

**Comment by ockham** [comment]
Update: A few PRs that we had originally planned for inclusion in WP 6.0.1 aren’t quite ready yet and/or were otherwise [deemed too risky](https://github.com/WordPress/gutenberg/pull/41807#issuecomment-1164560396), so we’ll be targetting WP 6.1 instead. They are:

- [Merge Comments Query Loop block into Post Comments block](https://github.com/WordPress/gutenberg/issues/40521) (existing [PR](https://github.com/WordPress/gutenberg/pull/41807)).
- Change the name and ID to Comments and `core/comments`.
  - Name (https://github.com/WordPress/gutenberg/pull/40763)
  - ID (https://github.com/WordPress/gutenberg/pull/40506)

I'll update the issue description accordingly.

Other than that, the [translation issue](https://github.com/WordPress/gutenberg/issues/41292) has an in-progress [PR](https://github.com/WordPress/wordpress-develop/pull/2826) that's slated to be included in WP 6.0.1. All other issues and/or PRs mentioned in this issue and the comments have been cherry-picked for inclusion in WP 6.0.1 🎉

**Comment by SantosGuillamot** [comment]
@c4rl0sbr4v0 I've seen that the [Comments translation issue](https://github.com/WordPress/gutenberg/issues/41292) has already been merged. I assume we can consider it done right?

**Comment by cbravobernal** [comment]
Yes! Tracking issue updated 🎉

**Comment by ockham** [comment]
Another small fix here: https://github.com/WordPress/gutenberg/pull/42131
I'll add it to the tracking issue.

**Comment by Poliuk** [comment]
@SantosGuillamot, now that 6.0.1 is out, I think it makes to close this issue and open a new one with the remaining tasks programmed for 6.1.

>  The following issues have been postponed to WP 6.1 (see https://github.com/WordPress/gutenberg/issues/41451#issuecomment-1165821623):
>
> - [Merge Comments Query Loop block into Post Comments block](https://github.com/WordPress/gutenberg/issues/40521) (existing [PR](https://github.com/WordPress/gutenberg/pull/40522)).
> - Change the name and ID to Comments and core/comments.
>   - Name (https://github.com/WordPress/gutenberg/pull/40763)
>   - ID (https://github.com/WordPress/gutenberg/pull/40506)

 Since you opened this one, I'll let you decide. What do you think?

**Comment by SantosGuillamot** [comment]
> now that 6.0.1 is out, I think it makes to close this issue and open a new one with the remaining tasks programmed for 6.1.

Mmm I'm not 100% sure if opening a new issue is needed, as there are currently only a few issues left, that are kind of related. Moreover, I feel the opening post and the discussion aren't too long yet. Would it make sense to keep using this tracking issue for 6.1? Honestly, I don't have a strong opinion here and both are fine for me.

**Comment by Poliuk** [comment]
> Would it make sense to keep using this tracking issue for 6.1?

Yeah, that should work as well, as long as we update this tracking issue, it should be fine 😊

**Comment by ockham** [comment]
Now that the Comments block ID change PR (#40506) is merged, the Comments/Post Comments blocks merge PR (#40521) is unblocked and has been rebased.

We now need to update a few things there (some basic stuff like `s/comments-query-loop/comments/g`) and give it a bit more testing.

**Comment by DAreRodz** [comment]
I've posted an update in https://github.com/WordPress/gutenberg/pull/41807#issuecomment-1179787698 regarding the block merge.

**Comment by DAreRodz** [comment]
Another update:

- I've added a button ([#41807 comment](https://github.com/WordPress/gutenberg/pull/41807#issuecomment-1180747046)) to switch from the legacy version to the "editable" version.
- I worked on some tweaks ([#42406](https://github.com/WordPress/gutenberg/pull/42406)) after migrating the Comments block's e2e tests to Playwright.

I will work on adding the different test cases the block now has after the merge. 🙂

**Comment by ockham** [comment]
@DAreRodz and I have continued to work on the Block merge PR (#41807) for the past couple of days. We

- removed a now-obsolete block deprecation warning (on the server side)
- changed the way we remove the `legacy` block attribute by using `setAttributes` rather than replacing the monolithic legacy block with a freshly created modular one
- de-duplicated some of the warnings the user would get if comments were disabled -- now only shown in lieu of the Post Comments Form block (rather than the entire Comments block)
- made some tweaks to the e2e tests to be more semantic

Pretty much all feedback has been addressed. I've already given approval; there was [one more piece of feedback](https://github.com/WordPress/gutenberg/pull/41807/files#r901484283) that David was looking into. We might either address that as part of this PR or in a follow-up. In any case, we're planning to merge today 🤞

**Comment by DAreRodz** [comment]
I've just merged https://github.com/WordPress/gutenberg/pull/41807 and left some [notes](https://github.com/WordPress/gutenberg/pull/41807#issuecomment-1192804470) for a follow-up PR. 😊

**Comment by annezazu** [comment]
Does this issue need to remain open? Seems the prior items listed for 6.1 have all been merged :)  https://github.com/WordPress/gutenberg/issues/41451#issuecomment-1176358358

**Comment by SantosGuillamot** [comment]
I was waiting until 6.1 just in case some bugs arise while testing, but maybe it makes sense to close it and open new issues if needed. I'm closing it now, thanks for the heads up! 🙂
